TLM/TLM/UI/TrafficManagerTool.cs

```diff
@@ -605,36 +825,41 @@ public struct NodeVisitItem {
-                segmentInput.m_netService.m_service = ItemClass.Service.Road;
-                segmentInput.m_ignoreTerrain = true;
-                segmentInput.m_ignoreSegmentFlags = NetSegment.Flags.None;
-                //segmentInput.m_ignoreSegmentFlags = NetSegment.Flags.Untouchable;
-
-                RaycastOutput segmentOutput;
-                if (RayCast(segmentInput, out segmentOutput)) {
+                var segmentInput = new RaycastInput(m_mouseRay, m_mouseRayLength) {
+                    m_netService = {
+                        // find road segments
+                        m_itemLayers = ItemClass.Layer.Default | ItemClass.Layer.MetroTunnels,
+                        m_service = ItemClass.Service.Road
+                    },
+                    m_ignoreTerrain = true,
+                    m_ignoreSegmentFlags = NetSegment.Flags.None
+                };
+                // segmentInput.m_ignoreSegmentFlags = NetSegment.Flags.Untouchable;
+
+                if (RayCast(segmentInput, out RaycastOutput segmentOutput)) {
                     HoveredSegmentId = segmentOutput.m_netSegment;
                 } else {
                     // find train segments
-                    segmentInput.m_netService.m_itemLayers = ItemClass.Layer.Default | ItemClass.Layer.MetroTunnels;
+                    segmentInput.m_netService.m_itemLayers =
+                        ItemClass.Layer.Default | ItemClass.Layer.MetroTunnels;
                     segmentInput.m_netService.m_service = ItemClass.Service.PublicTransport;
-                    segmentInput.m_netService.m_subService = ItemClass.SubService.PublicTransportTrain;
+                    segmentInput.m_netService.m_subService =
+                        ItemClass.SubService.PublicTransportTrain;
                     segmentInput.m_ignoreTerrain = true;
                     segmentInput.m_ignoreSegmentFlags = NetSegment.Flags.None;
-                    //segmentInput.m_ignoreSegmentFlags = NetSegment.Flags.Untouchable;
+                    // segmentInput.m_ignoreSegmentFlags = NetSegment.Flags.Untouchable;
 
                     if (RayCast(segmentInput, out segmentOutput)) {
                         HoveredSegmentId = segmentOutput.m_netSegment;
                     } else {
                         // find metro segments
-                        segmentInput.m_netService.m_itemLayers = ItemClass.Layer.Default | ItemClass.Layer.MetroTunnels;
+                        segmentInput.m_netService.m_itemLayers =
+                            ItemClass.Layer.Default | ItemClass.Layer.MetroTunnels;
                         segmentInput.m_netService.m_service = ItemClass.Service.PublicTransport;
-                        segmentInput.m_netService.m_subService = ItemClass.SubService.PublicTransportMetro;
-                        segmentInput.m_ignoreTerrain = true;
+                        segmentInput.m_netService.m_subService =
+                            ItemClass.SubService.PublicTransportMetro;
```

Comment:
same here (`m_ignoreTerrain`)

Reply:
Same, it is set to true some lines above and never changed since.